components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/JsExecutable.java

```diff
@@ -104,15 +123,24 @@ protected String getRelativePath() {
 
     @Override
     public Object execute(Object context, API api) {
+        String functionName = null;
         try {
             engineBindings.setJSFunctionProvider(new JsFunctionsImpl(api));
-            return engine.invokeFunction("onRequest", context);
+            functionName = api.getRequestLookup().getRequest().isGet() ? FUNCTION_ON_GET : FUNCTION_ON_POST;
+            if (functionName.equals(FUNCTION_ON_GET) && isOnGetFound) {
+                return engine.invokeFunction(functionName, context);
+            } else if (functionName.equals(FUNCTION_ON_POST) && isOnPostFound) {
+                return engine.invokeFunction(functionName, context);
+            } else {
+                return null;
+            }
         } catch (ScriptException e) {
-            throw new UUFException("An error occurred when executing the 'onRequest' function in JavaScript file '" +
-                                           absolutePath + "' with context '" + context + "'.", e);
+            throw new UUFException(
+                    "An error occurred when executing the '" + functionName + "' function in JavaScript file '" +
+                            absolutePath + "' with context '" + context + "'.", e);
         } catch (NoSuchMethodException e) {
```

Comment:
We need to catch the `NoSuchMethodException` which is thrown by `engine.invokeFunction()` method. So we need this `catch`.